### PR TITLE
V0.1.0/authentication - Add component Authentication

### PR DESCRIPTION
**Pull request by SpQuyt**
go pro thoi giao su

### CODE REVIEW

File: src/components/Authentication/InputField/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React, { Component } from 'react';
+import {
+  View, TextInput,
+} from 'react-native';
+import styles from './index.styles';
+
+class InputField extends Component {
+  render() {
+    const { placeholder, setText, isSecureText } = this.props;
```
Comment (namdaoduy) [inline src/components/Authentication/InputField/index.js]: props name ** setText** should be ** onChangeText** => more clearly

Comment (namdaoduy) [inline src/components/Authentication/InputField/index.js]: props name ** setText** should be ** onChangeText** => more clearly

File: src/components/Authentication/InputField/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    return (
+      <View style={styles.inputField}>
+        <TextInput
+          placeholder={placeholder}
+          onChangeText={(text) => setText(text)}
```
Comment (namdaoduy) [inline src/components/Authentication/InputField/index.js]: same as: `onChangeText={onChangeText}`

File: src/components/Authentication/Login/index.js [hunk 3]
```diff
@@ -1,30 +1,96 @@
...
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      loading: false,
+    };
+  }
+
+  // If I remove this method it will make error of "Unused text fields" on the inputForm
+  getInitialState = () => {
```
Comment (namdaoduy) [inline src/components/Authentication/Login/index.js]: Remove this. Not prefer "dummy codes" to disable eslint warning

File: src/components/Authentication/Login/index.js [hunk 4]
```diff
@@ -1,30 +1,96 @@
...
+          <Image
+            style={styles.imgLogo}
+            source={logoPath}
+          />
+        </View>
+
+        <View style={styles.formContainer}>
+          <InputField
+            placeholder="Email..."
+            setText={(text) => this.setState({ email: text })}
```
Comment (namdaoduy) [inline src/components/Authentication/Login/index.js]: Write a method: `onChangeText`
Reference: https://stackoverflow.com/questions/44416541/react-native-difference-between-onchange-vs-onchangetext-of-textinput

File: src/components/Authentication/Login/index.js [hunk 5]
```diff
@@ -1,30 +1,96 @@
...
+        <View style={styles.imgContainer}>
+          <Image
+            style={styles.imgLogo}
+            source={logoPath}
+          />
+        </View>
+
+        <View style={styles.formContainer}>
+          <InputField
+            placeholder="Email..."
```
Comment (namdaoduy) [inline src/components/Authentication/Login/index.js]: Is `value` field missing?

File: src/components/Authentication/Login/index.js [hunk 6]
```diff
@@ -1,30 +1,96 @@
...
+          />
+          <InputField
+            placeholder="Password..."
+            setText={(text) => this.setState({ password: text })}
+          />
+        </View>
+
+        <View style={styles.buttonContainer}>
+          {this.renderLoginButton()}
+          <TouchableOpacity
```
Comment (namdaoduy) [inline src/components/Authentication/Login/index.js]: Why use `renderLoginButton` for the LOGIN, but not for the `change to sign up`

File: src/components/Authentication/Login/index.js [hunk 7]
```diff
@@ -1,30 +1,96 @@
...
+            style={[styles.button, { backgroundColor: 'transparent' }]}
```
Comment (namdaoduy) [inline src/components/Authentication/Login/index.js]: Create another style only for this component

File: src/datalayer/actions/auth.action.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import { authAction } from 'constants/actions';
+
+export const changeToLoginComponent = () => ({
+  type: authAction.IS_LOGIN_COMPONENT,
+});
+
+export const changeToSignupComponent = () => ({
```
Comment (namdaoduy) [inline src/datalayer/actions/auth.action.js]: `navigateToSignup` would be great

File: src/components/Authentication/Login/index.styles.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  },
+  button: {
+    padding: 10,
+    margin: 5,
+    backgroundColor: '#fb3',
+    alignItems: 'center',
+    marginHorizontal: 5,
+    borderRadius: 8,
+  },
+  buttonContainer: {
```
Comment (namdaoduy) [inline src/components/Authentication/Login/index.styles.js]: container go before inner

File: src/components/Authentication/SignUp/index.js [hunk 10]
```diff
@@ -1,25 +1,112 @@
...
 
-const styles = StyleSheet.create({
-  container: {
-    flex: 1,
-    justifyContent: 'center',
-    alignItems: 'center',
-    backgroundColor: '#F5FCFF',
-  },
-});
+// Import redux
```
Comment (namdaoduy) [inline src/components/Authentication/SignUp/index.js]: Redundant comment

File: src/components/Authentication/SignUp/index.js [hunk 11]
```diff
@@ -1,25 +1,112 @@
...
+      fullname: '',
+      email: '',
+      password: '',
+      confirmPassword: '',
+      loading: false,
+    };
+  }
+
+  // If I remove this method it will make error of "Unused text fields" on the inputForm
+  getInitialState = () => {
```
Comment (namdaoduy) [inline src/components/Authentication/SignUp/index.js]: Remove this

File: src/components/Authentication/SignUp/index.js [hunk 12]
```diff
@@ -1,25 +1,112 @@
...
+          <Image
+            style={styles.imgLogo}
+            source={logoPath}
+          />
+        </View>
+
+        <View style={styles.formContainer}>
+          <InputField
+            placeholder="Fullname..."
+            setText={(text) => this.setState({ email: text })}
```
Comment (namdaoduy) [inline src/components/Authentication/SignUp/index.js]: Use `onChangeText`, same as Login

File: src/components/Authentication/index.js [hunk 13]
```diff
@@ -1,29 +1,28 @@
...
 
-const styles = StyleSheet.create({
-  container: {
-    flex: 1,
-    justifyContent: 'center',
-    alignItems: 'center',
-    backgroundColor: '#F5FCFF',
-  },
-});
+// Import redux
```
Comment (namdaoduy) [inline src/components/Authentication/index.js]: Redundant. (Make sure you fix some review that appear for both Login and Signup)

File: src/datalayer/reducers/auth.reducer.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+  isLoginComponent: true,
+};
+
+export default (state = INITIAL_STATE, action) => {
+  console.log(state.isLoginComponent);
+  switch (action.type) {
+    case authAction.IS_LOGIN_COMPONENT: {
+      return {
+        ...state,
+        isLoginComponent: true,
```
Comment (namdaoduy) [inline src/datalayer/reducers/auth.reducer.js]: Not recommend (can not be extended). Here're some other solutions:
- `component`: "login" | "signup" | ...
- ...

File: src/datalayer/reducers/auth.reducer.js [hunk 15]
```diff
@@ -0,0 +1,27 @@
+import { authAction } from 'constants/actions';
+
+export const INITIAL_STATE = {
+  isLoginComponent: true,
```
Comment (namdaoduy) [inline src/datalayer/reducers/auth.reducer.js]: Add comment to describe the meaning of this state

File: src/components/Authentication/Login/index.js (hunk 6 above)
Comment (SpQuyt) [reply src/components/Authentication/Login/index.js]: Cuz clicking `loginButton` needs time to connecting API from server. `change to sign up` doesnt need that

File: src/datalayer/actions/auth.action.js (hunk 8 above)
Comment (namdaoduy) [reply src/datalayer/actions/auth.action.js]: Sure

File: src/components/Authentication/Login/index.js (hunk 6 above)
Comment (namdaoduy) [reply src/components/Authentication/Login/index.js]: Ok

**Pull request merged**